File: src/main/java/com/example/wantedmarketapi/controller/MemberController.java
package com.example.wantedmarketapi.controller;

import com.example.wantedmarketapi.common.BaseResponse;
import com.example.wantedmarketapi.converter.MemberConverter;
import com.example.wantedmarketapi.dto.request.MemberRequestDto.*;
import com.example.wantedmarketapi.dto.response.MemberResponseDto.*;
import com.example.wantedmarketapi.exception.GlobalErrorCode;
import com.example.wantedmarketapi.service.MemberCommandService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("api/v1/members")
@Tag(name = "😎 Member", description = "사용자 관련 API")
public class MemberController {

    private final MemberCommandService memberCommandService;

    @Operation(summary = "회원가입 API", description = "회원가입을 진행합니다")
    @ApiResponses({
            @ApiResponse(responseCode = "201", description = "성공"),
    })
    @PostMapping("/signup")
    @ResponseStatus(HttpStatus.CREATED)
    public BaseResponse<SignUpMemberResponse> signUpMember(@RequestBody SignUpMemberRequest request) {
        return BaseResponse.onSuccess(
                GlobalErrorCode.CREATED,
                MemberConverter.toSignUpMemberResponse(memberCommandService.signUpMember(request)));
    }
}

File: src/main/kotlin/wanted/market/api/member/presentation/MemberController.kt
package wanted.market.api.member.presentation

import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.responses.ApiResponse
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import wanted.market.api.common.ApiResultResponse
import wanted.market.api.member.application.MemberService
import wanted.market.api.member.domain.dto.out.RetrieveMemberResult

@RestController
@RequestMapping("/members")
@Tag(name = "02. 회원 API", description = "인증 또는 권한 부여 후에 접근할 수 있는 회원 API")
class MemberController(
    private val memberService: MemberService
) {

    @Operation(
        summary = "모든 회원 조회", responses = [
            ApiResponse(responseCode = "200", description = "모든 회원 조회 성공")
        ]
    )
    @GetMapping
    fun findAllMembers(): ApiResultResponse<List<RetrieveMemberResult>> {
        return ApiResultResponse(
            data = memberService.findAllMembers()
        )
    }
}